pkg/cdi/cache.go

```diff
@@ -133,11 +132,11 @@ func (c *Cache) Refresh() error {
 	}
 
 	// collect and return cached errors, much like refresh() does it
-	var result error
+	errs := []error{}
 	for _, errors := range c.errors {
-		result = multierror.Append(result, errors...)
+		errs = append(errs, errors...)
 	}
-	return result
+	return errors.Join(errs...)
```

Comment:
var errs error
for _, errors := range c.errors {
    errs = errors.Join(errs, errors)
}
```

Reply:
Yeah, I did this wrong, as it resulted in a flat topology, whereas a joinError of one joinErrors per Spec file wrapping errors of that file only is more natural. Updated like that.